extensions/ql-vscode/src/query-history.ts

```diff
@@ -488,19 +540,21 @@ export class QueryHistoryManager extends DisposableObject {
     singleItem: FullQueryInfo,
     multiSelect: FullQueryInfo[]
   ): Promise<void> {
-    if (!this.assertSingleQuery(multiSelect)) {
+    const { finalSingleItem, finalMultiSelect } = this.determineSelection(singleItem, multiSelect);
+
+    if (!this.assertSingleQuery(finalMultiSelect)) {
```

Comment:
Thanks for pointing it out, and for making the fix! 

In the future, it'd personally help me a lot of we had separate PRs for such changes. I can live with it if it's too much trouble, I just find that I miss stuff more in bigger PRs that do multiple things.

Reply:
In general, I agree and prefer that for reviewing. But for larger changes like these, where nothing is working (until everything is), I find it quite hard to break out these smaller bits, especially since these bits are often somewhat dependent on the larger change. Sometimes, I try separating these out into individual commits in the same PR.

Reply:
I think it's still ok to separate things out in separate PRs even when you work on a larger bit of work. 

If I'm working on something fairly big I start off the work and if I find something that could be split into its own PR then I do that. It's a bit more hassle juggling things but I find that 
a) it's easier for the reviewer 
b) things get done more quickly overall and 
c) there's less of a chance that things get missed

I'm probably "preaching to the converted" and apologies for that, just wanted to make it clear what my personal preferences (and weaknesses!) are as a reviewer.